packages/components/src/components/hds/icon/index.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { guidFor } from '@ember/object/internals';
+import { assert } from '@ember/debug';
+import { iconNames } from '@hashicorp/flight-icons/svg';
+
+import type { IconName } from '@hashicorp/flight-icons/svg';
+
+export interface HdsIconSignature {
+  Args: {
+    name?: IconName;
+    color?: string;
+    size?: '16' | '24';
+    stretched?: boolean;
+    isInline?: boolean;
+    title?: string;
+  };
+  Element: SVGElement;
+}
+
+export default class HdsIcon extends Component<HdsIconSignature> {
+  constructor(owner: unknown, args: HdsIconSignature['Args']) {
+    super(owner, args);
+
+    if (!this.args.name) {
+      assert('Please provide to <Hds::Icon> a value for @name');
+    } else if (!iconNames.includes(this.args.name)) {
+      assert(
+        `The icon @name "${this.args.name}" provided to <Hds::Icon> is not correct. Please verify it exists on https://helios.hashicorp.design/icons/library`
+      );
+    }
+  }
+
+  get isInline() {
+    return this.args.isInline ?? false;
+  }
+
+  /**
+   * Sets the color for the SVG
+   *
+   * @param color {string}
+   * @default 'currentColor'
+   */
+  get color() {
+    return this.args.color ?? 'currentColor';
+  }
+
+  /**
+   * Generates a unique ID for the SVG
+   *
+   * @param iconId
+   */
+  iconId = 'icon-' + guidFor(this);
+
+  /**
+   * Indicates which icon should be used. An error (in the form of an assertion)
+   * will occur if a value has not been provided.
+   *
+   * @param name {string}
+   */
+  get name() {
+    return this.args.name;
+  }
+
+  /**
+   * Gets the icon's size (16 or 24)
+   *
+   * @param size
+   * @returns the value of `size` if set
+   * @default `16`
+   */
+  get size() {
+    return this.args.size ?? '16';
```

Comment:
question: should we reference the `16` value here from an enum (see my other comment)?